.pre-commit-config.yaml

```diff
@@ -18,6 +18,5 @@ repos:
         pass_filenames: false
         args: ["--config-file=pyproject.toml", "src"]
         additional_dependencies:
-          - attrs
```

Comment:
Can you share info on the benefit of switching?  Early on we made a decision to utilize `attrs` - and not use `dataclass` - in both `somacore` and `tiledbsoma`, so it is definitely present for any environment where `tiledbsoma-ml` is in use.  Without a concrete benefit, my instinct is to maintain that consistency as this package is part of the same ecosystem (ie.., same maintainers).

Reply:
Discussed offline, we use `kw_only` in this project, which only arrived in `dataclasses` in 3.10 (which failed CI for <3.10), so `attrs` is definitely necessary for now.